activity/permissions.py

```diff
@@ -0,0 +1,16 @@
+from rest_framework import permissions
+
+from workflow.models import ActivityUserOrganizationGroup
+
+
+class IsReadOnly(permissions.BasePermission):
+    def has_permission(self, request, view):
+        if request.method in permissions.SAFE_METHODS:
+            return True
+
+        return not ActivityUserOrganizationGroup.objects.filter(
```

Comment:
Need to check whether the user is authenticated.

Reply:
I would rather IsAuthenticated let handle it.

Reply:
It is a generic permission. So you can use this permission in other view where you have not used `IsAuthenticated`. So in that case if any unauthenticated user try to make request like `post` or `put`, it will throw 500

Reply:
@safwanrahman I think those two checks should be mutually exclusive.

Reply:
Having them in separate classes makes it easier to add unit tests.

Reply:
@michaelbukachi I understand that. But in that case you should check `user.is_authenticated`. Otherwise if you use this permission class where you have not used `IsAuthenticated`, it will raise 500 error.

Reply:
I see. @nasirhjafri could you make the adjustments?

Reply:
Either way, we are using it with IsAuthenticated or not I think we should add a check for `user.is_authenticated`.